sql/core/src/test/scala/org/apache/spark/sql/TPCDSCollationQueryTestSuite.scala

```diff
@@ -0,0 +1,270 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+import java.nio.file.{Files, Paths}
+import java.util.Locale
+
+import org.apache.spark.{SparkConf, SparkContext}
+import org.apache.spark.sql.catalyst.util.resourceToString
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.test.TestSparkSession
+import org.apache.spark.tags.ExtendedSQLTest
+import org.apache.spark.util.Utils
+
+/**
+ * End-to-end tests to validate TPC-DS query results against collation-aware
+ * modified data and queries.
+ *
+ * For each collation, table schemas are replicated into two databases in such way that in first
+ * DB all table columns are collated with specified collation, while the second DB collates table
+ * columns with case-insensitive version of the collation. Tables from first DB are then populated
+ * with lowercase-converted data from tpc-ds kit and tables from second DB are populated with
+ * randomized-case data.
+ *
+ * When running arbitrary SQL query, we convert the query to lowercase for execution
```

Comment:
convert the query to lowercase?

Reply:
Added clarification to the comment, please check.

Reply:
What does it mean to convert query to lowercase? Lowercase the full query string?

Reply:
That's right. We lowercase the full query text contents and in this way string literals will end up being lowercase which is required to produce the same results as case-insensitive collation.

For example, if the query is something like `select count(*) where s = 'A'`, when converted to lowercase, query will count number of rows where we have "a" instead of "A" to produce correct results on row data which is already in lowercase. We need this in order to be able to compare the results with the same query executed against collated table.

Can add more clarification to the header comment and include this example if you think it's still unclear.

Reply:
We should highlight that it's for lowercasing the string literals in the query string.